server/src/index.ts

```diff
@@ -48,23 +54,48 @@ io.on("connection", function(socket) {
 
   socket.on("new-player", (p: NewPlayer) => {
     Player.Players[p.id] = new Player(socket, p);
-
     const game = gameByPlayer(p.id);
 
     if (game) {
       game.addPlayer(p.id, socket);
       socket.emit("joined-game", game.id);
     }
 
+    if (_disconnections[p.id]) {
+      console.log(`player ${p.id} has reconnected back`);
+
+      clearTimeout(_disconnections[p.id]);
+      delete _disconnections[p.id];
+      
+      game.removeDisconnectedPlayer(p.id);
+    } else {
+      console.log(`new playered with id ${p.id} joined the game!`);
+    }
+
     updatePlayers();
   });
 
   socket.on("disconnect", () => {
+    // onPlayerDisconnect
     const p = playerBySocket(socket.id);
 
     if (!p) {
       return;
     }
+    
+    const game = gameByPlayer(p.id);
```

Comment:
what if player is not in game?

Reply:
than player will suck someone's dick hahaha